### ISSUE TITLE
Target .NET Core and Version 1.1.0

### ISSUE THREAD

**Issue by stevejgordon** [HTBox/allReady#1637]
To enable Mac developers to work on this project it would be nice to move over to targeting the .NET core framework.

@tonysurma - I believe some work was fairly close for that?

I can see one potential issue where we rely on System.Drawing to generate a QR Code for the mobile app. I'm not sure if that functionality is used / needed at the moment? If we need this, then we are tied to the full framework or need to look at any .NET core libraries that may be available.

At the same time, perhaps we review moving to the 1.1.0 release train to get the latest features and improvements?

Throwing this out as a question so we can determine if this should be addresses pre v1. At least the Mac support would be nice. I have one Mac user coming to the code-a-thon in January at NDC so would be good to have support for it if possible.

**Comment by tonysurma** [comment]
@stevejgordon lets remove that code and dependency.  It is not needed/used and it would be great to move to .Net Core ASAP

I have been pushing/trying the Mac support for ages so all for it.

**Comment by stevejgordon** [comment]
@tonysurma Cool, once we get through the Dec milestone pieces I'm all for trying to get this done. I'm hoping it'll be reasonably smooth. It'll be a good next blog post as well!

**Comment by VishalMadhvani** [comment]
FYI: https://icanhasdot.net/result?github=HTBox~2FallReady

**Comment by stevejgordon** [comment]
I have this underway in a branch which I'll work on around higher priority issues

**Comment by stevejgordon** [comment]
Might be a bigger job than I appreciated as quite a few libraries (as @VishalMadhvani's link highlights) are not .net core ready.

We can potentially work around some of that by wrapping the APIs (Twilio/SendGrid etc) directly and not using their c# libraries. For the features we need, this might be fine.

I'll continue digging as I need to research each dependency and see what can be used and for those that can't, how tied to them we are.

**Comment by stevejgordon** [comment]
Update - Spent a little more time with this today and slowly progressing it. I've managed to update / remove a few problem dependencies. There are a few that will require a little more review though.

**Comment by stevejgordon** [comment]
I believe the remaining dependencies that cause problems are the GeoCoding.net library and the LinqToTwitter library.

We already have discussed GeoCoding.net since we're using it just to get lat/long from addresses. As we don't take advantage of anything else the library offers, it will be best to create our own wrapper around the Google API to handle any address geocoding we need to do. I'm currently working on that in my .NET core branch.

LinqToTwitter seems to be just used for getting a few Twitter account details when we authenticate using Twitter. Again, this looks to wrap an API so while I haven't looked too deep, I believe it can be replaced with our own wrapper to call the API directly. They do have a PR accepted which is meant to add .net core support. However they have not updated their NuGet package at the time of writing this update.

@mgmccarthy - I think you may have worked on the Twitter stuff a while back. Any concerns with potential removal of this library assuming we can use APIs to get the data.

**Comment by mgmccarthy** [comment]
@stevegordon, don't bother. I already have a working invocation of google maps using HttpClient removing IGeoCoder on a local branch!

**Comment by mgmccarthy** [comment]
@stevegordon, I can look into getting rid of Linq2Twitter and invoicing API directly. It's only used to support sign-on the site using Twitter external auth provider.

**Comment by VishalMadhvani** [comment]
ohhhh this sounds very promising @stevejgordon

**Comment by stevejgordon** [comment]
@mgmcarthy I have the bare bones code to allow removal of linqtotwitter started.

**Comment by stevejgordon** [comment]
@mgmccarthy - It's not tested but I have initial code for both. I'll try to get a WIP PR in soon and we can chat around what I've included. As I have replaced the dependencies that couldn't be used I'm now compiling a bit further and have found a few more code areas to change due to differences in API surface in .net core. So I'll look to tweak those also before putting in an initial PR. I might get some time today at lunchtime or if not, hopefully tomorrow evening.

I'm hoping to have this up and running before the NDC code-a-thon if at all possible.
